javascript/javascript1/week2/Week2.js

```diff
@@ -0,0 +1,159 @@
+    }
+    return todaysDress;
+}
+const clothesToWear = dressAdviser(9);
+console.log(clothesToWear);
+
+/*Candy helper optional */
+
+const availibleCandyTypes = [];
+
+const boughtCandyPrices = [];
+
+function addCandy(candyType, weight) {
+    let price;
+    switch (candyType) {
+        case 'Sweet':
+            price = weight * 0.5;
+            break;
+        case 'Chocolate':
+            price = weight * 0.7;
+            break;
+        case 'Toffee':
+            price = weight * 1.1;
+            break;
+        case 'Chewing-gum':
+            price = weight * 0.03;
+            break;
+        default:
+            price = " price not set";
+            break;
+    }
+    availibleCandyTypes.push(candyType, price);
+    boughtCandyPrices.push(price);
+}
+
+addCandy("Sweet", 20);
+addCandy("Chocolate", 40);
+addCandy("Toffee", 30);
+addCandy("Chewing-gum", 100);
+console.log(availibleCandyTypes);
+console.log(boughtCandyPrices);
+
+const amountToSpend = (Math.random() * 100).toFixed(); // gives any number betweeen o to 100
+function canBuyMoreCandy() {
+
+    let cost = 0;
+
+    //    for (let i = 0; i < boughtCandyPrices.length; i++) {
+    //        console.log(boughtCandyPrices[i]);
+    //         cost += boughtCandyPrices[i];
+    //     }
+
+
+    // alternate way : rewriting the above for-loop into while-loop
+    let i = 0;
+    while (i < boughtCandyPrices.length) {
+        cost += boughtCandyPrices[i];
+        i++;
+    }
+
+    console.log(`Amount to spend is : ${ amountToSpend } and  cost is :  ${cost}`);
+
+    if (amountToSpend <= cost) {
+        console.log(`you already have ${(cost - amountToSpend)}  to pay back so no more candy for you!`);
+    } else {
+        console.log(` you still have ${amountToSpend - cost}  left soYou can buy more, so please do!`);
+    }
+}
+
+canBuyMoreCandy()
+
+/*Student Manager*/
+const class07StudentsArray = [];
+function addStudentToClass(studentName) {
+
+  let admission;
+  if (class07StudentsArray.length < 6) {
+    class07StudentsArray.push(studentName);
+    admission = `Number of students in class now is ${class07StudentsArray.length} . Congratulation ${studentName} ! you are admitted`;
+    console.log(admission);
+  } else if (class07StudentsArray.includes(studentName)) {
+    admission = `Hi ${studentName} you are already admitted.`;
+    console.log(admission);
+  } else if (class07StudentsArray.length <= 6 && studentName === "Queen") {
+    class07StudentsArray.push(studentName);
+    admission = `Honourable ${studentName} is successfully added to the class07. The class member now including queen is ${class07StudentsArray.length}`;
+    console.log(admission);
+  } else {
+    admission = `Sorry ${studentName} ! we have already ${class07StudentsArray.length} students admitted. The class is full, Cannot add more students to class 07`;
+    console.log(admission);
+  }
```

Comment:
There is a bug you need to consider which is cause by the flow of your if statements!
- If we have 3 students in the class, one of them is Vlad, and then we try to add Vlad again he will be wrongly admitted! (hint: it's because the < 6 statement is the first one, so if there is still space in the classroom no matter the name the student will be added)

Also, you can simplify this statement:
```if (class07StudentsArray.length <= 6 && studentName === "Queen")```
cause you don't care about the size of the classroom when adding the Queen, you only care if she's been already added!